src/integration-test/java/com/mongodb/jdbc/integration/testharness/TestGenerator.java

```diff
@@ -0,0 +1,145 @@
+        List<Boolean> expectedIsSigned = new ArrayList<>();
+        List<Boolean> expectedIsWritable = new ArrayList<>();
+
+        File directory = new File(GENERATED_TEST_DIR);
+        ResultSetMetaData resultSetMetadata = rs.getMetaData();
+        Map<String, Object> tests = new LinkedHashMap<String, Object>();
+        List<Map<String, Object>> testCases = new ArrayList<>();
+        Map<String, Object> testCase = new LinkedHashMap<String, Object>();
+        testCases.add(testCase);
+        tests.put("tests", testCases);
+
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        String fileName =
+                new SimpleDateFormat("'" + testEntry.description + "'MMddHHmmss'.yaml'")
+                        .format(new Date());
+
+        // generating 'expected_result'
+        ArrayList<List<Object>> result = new ArrayList<>();
+        while (rs.next()) {
+            List<Object> row = new ArrayList<>();
+            for (int i = 1; i <= resultSetMetadata.getColumnCount(); i++) {
+                row.add(rs.getObject(i));
+            }
+            result.add(row);
+        }
+
+        // generating expected resultset metadata
+        for (int i = 1; i <= resultSetMetadata.getColumnCount(); i++) {
+            expectedSqlType.add(TestTypeInfo.typesIntToString(resultSetMetadata.getColumnType(i)));
+            expectedCatalogName.add(resultSetMetadata.getCatalogName(i));
+            expectedColumnClassName.add(resultSetMetadata.getColumnClassName(i));
+            expectedColumnLabel.add(resultSetMetadata.getColumnLabel(i));
+            expectedColumnDisplaySize.add(resultSetMetadata.getColumnDisplaySize(i));
+            expectedPrecision.add(resultSetMetadata.getPrecision(i));
+            expectedScale.add(resultSetMetadata.getScale(i));
+            expectedSchemaName.add(resultSetMetadata.getSchemaName(i));
+            expectedIsAutoIncrement.add(resultSetMetadata.isAutoIncrement(i));
+            expectedIsCaseSensitive.add(resultSetMetadata.isCaseSensitive(i));
+            expectedIsCurrency.add(resultSetMetadata.isCurrency(i));
+            expectedIsDefinitelyWritable.add(resultSetMetadata.isDefinitelyWritable(i));
+            expectedIsNullable.add(
+                    TestTypeInfo.nullableIntToString(resultSetMetadata.isNullable(i)));
+            expectedIsReadOnly.add(resultSetMetadata.isReadOnly(i));
+            expectedIsSearchable.add(resultSetMetadata.isSearchable(i));
+            expectedIsSigned.add(resultSetMetadata.isSigned(i));
+            expectedIsWritable.add(resultSetMetadata.isWritable(i));
+        }
+
+        testCase.put("description", testEntry.description);
+        if (testEntry.meta_function != null) {
+            testCase.put("meta_function", testEntry.meta_function);
+        } else {
+            testCase.put("sql", testEntry.sql);
+        }
+        testCase.put("expected_result", result);
+        testCase.put("row_count", testEntry.row_count);
```

Comment:
Should this value also be generated? That way it conveniently updates if we ever update our datasets?